channeldb/graph.go

```diff
@@ -1631,11 +1664,21 @@ func delChannelByEdge(edges *bbolt.Bucket, edgeIndex *bbolt.Bucket,
 	}
 
 	// Finally, with the edge data deleted, we can purge the information
-	// from the two edge indexes.
+	// from the two edge indexes and mark the edge as a zombie.
 	if err := edgeIndex.Delete(chanID); err != nil {
 		return err
 	}
-	return chanIndex.Delete(b.Bytes())
+	if err := chanIndex.Delete(b.Bytes()); err != nil {
+		return err
+	}
+
+	var pubKey1, pubKey2 [33]byte
+	copy(pubKey1[:], nodeKeys[:33])
+	copy(pubKey2[:], nodeKeys[33:])
+
+	return markEdgeZombie(
```

Comment:
We may want to distinguish between _deleting_ and edge, and marking it as a zombie...

Reply:
Fixed.